suite-native/graph/src/components/TimeSwitchItem.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { TouchableOpacity } from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Text } from '@suite-native/atoms';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { getSelectedTimeFrame, setSelectedTimeFrame } from '../reducer';
+import { TimeFrameValues } from '../reducers/types';
+
+type TimeSwitchItemProps = {
+    value: TimeFrameValues;
+    shortcut: string;
+};
+
+type TextStyleProps = {
+    isSelected: boolean;
+};
+const textStyle = prepareNativeStyle<TextStyleProps>((utils, { isSelected }) => ({
+    ...utils.typography.hint,
+    color: isSelected ? utils.colors.forest : utils.colors.gray500,
+    textTransform: 'uppercase',
+}));
+
+export const TimeSwitchItem = ({ value, shortcut }: TimeSwitchItemProps) => {
+    const dispatch = useDispatch();
+    const { applyStyle } = useNativeStyles();
+    const selectedTimeFrame = useSelector(getSelectedTimeFrame);
```

Comment:
nit: Would be cool to keep the order of hooks used at least a bit consistent across the components perhaps? For example, I would always put selectors and state above everything.

Reply:
Yeah no problem. I like to use the "imports" (dispatch, styles, etc) on the top and then state and vars. But it makes more sense to have state on top I guess.

Reply:
https://github.com/trezor/trezor-suite/pull/5838/commits/8d03ac13bbdc855f35c585b400951d9764b552cb